<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="rgb_image_topic" default="xtion/rgb/image_rect_color"/>
  <arg name="rgb_camera_info_topic" default="xtion/rgb/camera_info"/>

  <arg name="depth_image_topic" default="xtion/depth/image_rect"/>
  <arg name="depth_camera_info_topic" default="xtion/depth/camera_info"/>

  <arg name="detector_model_filename" default="$(find uwds3_perception)/models/detection/ssd_mobilenet_v2_coco_2018_03_29.pb"/>
  <arg name="detector_weights_filename" default="$(find uwds3_perception)/models/detection/ssd_mobilenet_v2_coco_2018_03_29.pbtxt"/>
  <arg name="detector_config_filename" default="$(find uwds3_perception)/config/detection/mscoco_config.yaml"/>

  <arg name="face_detector_model_filename" default="$(find uwds3_perception)/models/detection/opencv_face_detector_uint8.pb"/>
  <arg name="face_detector_weights_filename" default="$(find uwds3_perception)/models/detection/opencv_face_detector.pbtxt"/>
  <arg name="face_detector_config_filename" default="$(find uwds3_perception)/config/detection/face_config.yaml"/>

  <arg name="shape_predictor_config_filename" default="$(find uwds3_perception)/models/estimation/shape_predictor_68_face_landmarks.dat"/>

  <arg name="use_gui" default="true"/>
  <arg name="use_depth" default="true"/>
  <arg name="use_faces" default="true"/>

  <node name="uwds3_perception" pkg="uwds3_perception" type="uwds3_perception_node.py" args="" output="screen" respawn="false">
    <rosparam subst_value="true">
      rgb_image_topic: $(arg rgb_image_topic)
      rgb_camera_info_topic: $(arg rgb_camera_info_topic)

      depth_image_topic: $(arg depth_image_topic)
      depth_camera_info_topic: $(arg depth_camera_info_topic)

      detector_model_filename: $(arg detector_model_filename)
      detector_weights_filename: $(arg detector_weights_filename)
      detector_config_filename: $(arg detector_config_filename)

      face_detector_model_filename: $(arg face_detector_model_filename)
      face_detector_weights_filename: $(arg face_detector_weights_filename)
      face_detector_config_filename: $(arg face_detector_config_filename)

      shape_predictor_config_filename: $(arg shape_predictor_config_filename)

      use_gui: $(arg use_gui)
      use_depth: $(arg use_depth)
      use_faces: $(arg use_faces)
    </rosparam>
  </node>

</launch>
